src/states/elder/mod.rs

```diff
@@ -405,32 +405,30 @@ impl Elder {
     fn complete_parsec_reset_data(&mut self, reset_data: ParsecResetData) -> CompleteParsecReset {
         let ParsecResetData {
             gen_pfx_info,
-            mut cached_events,
+            cached_events,
             completed_events,
         } = reset_data;
 
-        cached_events.extend(
-            self.parsec_map
-                .our_unpolled_observations()
-                .filter_map(|obs| match obs {
-                    parsec::Observation::OpaquePayload(event) => {
-                        if !completed_events.contains(&event.payload) {
-                            Some(event)
-                        } else {
-                            None
-                        }
-                    }
-
-                    parsec::Observation::Genesis { .. }
-                    | parsec::Observation::Add { .. }
-                    | parsec::Observation::Remove { .. }
-                    | parsec::Observation::Accusation { .. }
-                    | parsec::Observation::StartDkg(_)
-                    | parsec::Observation::DkgResult { .. }
-                    | parsec::Observation::DkgMessage(_) => None,
-                })
-                .cloned(),
-        );
+        let cached_events: BTreeSet<_> = cached_events
```

Comment:
just wondering why switch from the previous `extend` to the current `chain` ?

normally I will prefer not to use something like `let cached_events: BTreeSet<_> = cached_events`

Reply:
because the filter apply to both, not sure what you are suggesting?

Reply:
Ah, I got what you mean. 
just forgot my above comment then, my mis-reading of the code-block.